locksmithctl/daemon.go

```diff
@@ -309,8 +313,15 @@ func runDaemon() int {
 		strategy = StrategyBestEffort
 	}
 
+	// XXX: complain loudly if besteffort is used
+	if strategy == StrategyBestEffort {
+		dlog.Errorf("Reboot strategy %q is deprecated and will be removed in the future.", strategy)
+		dlog.Errorf("Please explicitly set the reboot strategy to one of %v", []string{StrategyOff, StrategyReboot, StrategyEtcdLock})
+		dlog.Error("See https://coreos.com/os/docs/latest/update-strategies.html for details on configuring reboot strategies.")
+	}
+
 	if strategy == StrategyOff {
-		fmt.Fprintf(os.Stderr, "Reboot strategy is %q - shutting down.\n", strategy)
+		dlog.Warningf("Reboot strategy is %q - shutting down.", strategy)
```

Comment:
Shouldn't be a warning, iirc only cloudinit tries to interpret strategy itself and mask the unit if it is off. We shouldn't warn the user because they aren't using our old cloudinit magic. notice should do fine.